### ISSUE TITLE
Being able to drag and adjust the order in which the bars appear in the BarView

### ISSUE THREAD

**Issue by dino-saurabh** [krharsh17/barview-android#73]
**Is your feature request related to a problem? Please describe.**
we arent able to drag and adjust the position of bars in the barView currently
**Describe the solution you'd like**
If we long press and then drag we should be able to adjust the position of the bars in the barView

**Comment by krharsh17** [comment]
This is a nice idea actually. But it would clash with #54 & #74 as all of these will be based on long clicks. Can you come up with something to avoid that? I'd love to have both the features in the repo 😄

**Comment by dino-saurabh** [comment]
@krharsh17 I have requested @Alaska18  to use swipe action for deletion since that would be more apt for deletion rather than adjustment 🙂

**Comment by krharsh17** [comment]
Well, for her set of features, there has to be a context menu that opens first, and then the user is allowed to choose from a list of options to either delete, fade or highlight a specific bar. Context menus are conventionally put on long-press events, so you'll have to think of a different approach for this.

**Comment by dino-saurabh** [comment]
Suggest me something ,I will also be looking for something that will work!!

**Comment by dino-saurabh** [comment]
Maybe add an option to swap pieces together in @Alaska18 's context menu and then implement this feature with a confirm button at the end of the workflow

**Comment by krharsh17** [comment]
Yup, that's a good way to do it. If you can come up with something till tomorrow, we'll consider that tomorrow, or if not, you can begin working on this. @SaumyaSingh1 @ShivangiSingh17 @samagragupta @vikasgola @sidhuparas Your suggestions would be helpful on this 🙂

**Comment by dino-saurabh** [comment]
![Screenshot_2020-03-29-16-22-33-316_com gg2 android apps youtube music](https://user-images.githubusercontent.com/30355336/77847279-a519e100-71d9-11ea-8b8b-3f6d7f424c03.jpg)
@krharsh17  something like this maybe having something to drag and adjust rows??

**Comment by sidhuparas** [comment]
@LuGO0 I was thinking about this drag and drop button just as you shown, but our app is a library so how can we place some kind of Edit button to show these drag buttons?

**Comment by dino-saurabh** [comment]
@sidhuparas what should we do for this reshuffling then any ideas!! Would love to hear em

**Comment by saumya1singh** [comment]
@LuGO0 ! Thankyou for putting your efforts 💯

I really loved your idea❤ I think the best way of tackling this is to show a context menu on long press as @krharsh17  suggested ✔


As said by @krharsh17 - 
`context menu that opens first, and then the user is allowed to choose from a list of options to either delete, fade or highlight a specific bar`

**Comment by sidhuparas** [comment]
@LuGO0 We might have to reconsider if dragging should be a feature since it might not fit our UI/UX. Do you have some strong reasons to implement this?

**Comment by Alaska18** [comment]
@LuGO0 I have a suggestion. When the user long clicks on the bar a menu will open up to select highlight fade or delete and when the user long clicks on label they can move they will he able to drag the bars.

**Comment by dino-saurabh** [comment]
@Alaska18 got your Idea but how do we confirm our reshuffled bars once we are done with adjustments ??

**Comment by Alaska18** [comment]
@LuGO0 I can suggest adding a checkbox to the side or at the bottom of whole barview. This checkbox will appear when the user long clicks the label of any bar and when the user is done rearranging the bars, He/she can check the checkbox after which the bars will get rearranged according to the user.

**Comment by Alaska18** [comment]
A text can be written next to the checkbox like "okay" or "done".:)

**Comment by krharsh17** [comment]
> @LuGO0 I have a suggestion. When the user long clicks on the bar a menu will open up to select highlight fade or delete and when the user long clicks on label they can move they will he able to drag the bars. 

This sounds good to me 👍🏻 Also, you could consider adding a 're-order' option in the menu to keep things simple and obvious.

> @LuGO0 I can suggest adding a checkbox to the side or at the bottom of whole barview. This checkbox will appear when the user long clicks the label of any bar and when the user is done rearranging the bars, He/she can check the checkbox after which the bars will get rearranged according to the user.

For apps that provide a top-level app bar, it'd be better if this confirm button could be integrated in it. [Here's](https://stackoverflow.com/questions/24567922/add-custom-actionbar-in-button-dynamically-in-android) a reference on how to achieve that. I am not sure how you can implement it here, but it'd be nice if this could be done. For apps that don't contain an app bar, a button or a checkbox as @Alaska18 suggested can be used.

**Comment by dino-saurabh** [comment]
Thanks everybody for suggestions I will start working on this now !!

**Comment by krharsh17** [comment]
@LuGO0 What's the status on this?

**Comment by dino-saurabh** [comment]
Yeah @krharsh17  I was busy with the other PRs now that they are merged I'll continue my work on this :)

**Comment by krharsh17** [comment]
@LuGO0 any updates on this?

**Comment by dino-saurabh** [comment]
Sorry @krharsh17 due to my sems resuming I couldnt do much you can unassign me and assign somebody else for this :(
